plugins/modules/nd_cluster_config_route.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2023, Shreyas Srish (@shrsr) <ssrish@cisco.com>
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "community"}
+
+DOCUMENTATION = r"""
+---
+module: nd_cluster_config_route
+short_description: Manages routes of the cluster config.
+description:
+- Manages the data and management routes of the cluster configuration.
+author:
+- Shreyas Srish (@shrsr)
+options:
+  destination_ip:
+    description:
+    - The destination address of the target network.
+    type: str
+  target_network:
+    description:
+    - Use 'data' to add a data network route.
+    - Use 'management' to add a management network route.
+    type: str
+    choices: [ data, management ]
+  state:
+    description:
+    - Use C(present) for adding a route to the cluster config.
+    - Use C(query) for listing all the routes of the cluster config.
+    - Use C(absent) for deleting a route of the cluster config.
+    type: str
+    choices: [ present, query, absent ]
+    default: present
+extends_documentation_fragment: cisco.nd.modules
+"""
+
+EXAMPLES = r"""
+- name: Create a data route
+  cisco.nd.nd_cluster_config_route:
+    destination_ip: 12.23.45.68/32
+    target_network: data
+    state: present
+
+- name: Query a route
+  cisco.nd.nd_cluster_config_route:
+    destination_ip: 12.23.45.68/32
+    state: query
+  register: query_result
+
+- name: Query all the routes
+  cisco.nd.nd_backup_restore:
+    state: query
+  register: query_results
+
+- name: Delete a route
+  cisco.nd.nd_backup_restore:
+    destination_ip: 12.23.45.68/32
+    state: absent
+"""
+
+RETURN = r"""
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.nd.plugins.module_utils.nd import NDModule, nd_argument_spec, sanitize_dict
+
+try:
+    from ansible.module_utils.six.moves.urllib.parse import quote
+
+    HAS_QUOTE = True
+except Exception:
+    HAS_QUOTE = False
+
+MATCH_MAPPING = dict(
+    data="Data",
+    management="Management",
+)
```

Comment:
Do we need this mapping when all we do is capitalising the input?

```
target_network = nd.params.get("target_network")
if target_network:
    target_network = target_network.capitalize()
```

OR single line

`target_network = nd.params.get("target_network").capitalize() if nd.params.get("target_network") else nd.params.get("target_network")`

Reply:
if we do need I think we should move to constants.py